cucumber-expressions/ruby/lib/cucumber/cucumber_expressions/parameter_type.rb

```diff
@@ -6,7 +6,7 @@ class ParameterType
       ILLEGAL_PARAMETER_NAME_PATTERN = /([\[\]()$.|?*+])/
       UNESCAPE_PATTERN = /(\\([\[$.|?*+\]]))/
 
-      attr_reader :name, :type, :regexps
+      attr_reader :name, :type, :regexps, :force_full_word
```

Comment:
I hope I'm commenting on the right line.

I don't think this should be a flag. I would consider the old behavior to be just wrong so no need to enable it.

Reply:
I'm wondering about (which is the reason why so far only ruby is implemented).

I think it could be helpful for advanced users who may want to finely tweak the way suggestions are done.
But in the other hand those people would be mature enough to know how to tweak the step definitions and not blindly copy-paste what has been suggested by the tool.

Well, I'm a bit lost here :D

Reply:
In Java this would mean yet another constructor variation. This makes the API more difficult to use because (when your IDE isn't perfectly) configured you'd have to guess between `....boolean, boolean)` and `...boolean, boolean, boolean)`. So I'd prefer not to have an extra one. 

Additionally the cucumber expression generator is already prone to producing combinatorial explosions. So anything that prunes the suggestions to include only the most likely ones is good.

Reply:
I agree with @mpkorstanje - we should not make this a configurable flag. YAGNI, KISS etc.

Reply:
that's been removed.